test/core/census/resource_test.c

```diff
@@ -136,22 +133,34 @@ static void test_base_resources() {
 }
 
 int main(int argc, char **argv) {
+  char *resource_empty_name_pb, *resource_full_pb, *resource_minimal_good_pb,
+  *resource_no_name_pb, *resource_no_numerator_pb, *resource_no_unit_pb;
+  if (argc >= 7) {
```

Comment:
I think better would be something like:

if (argc == 7) {
   set params from args...;
} else if (argc == 1) {
   set params to explicit defaults;
} else {
  fail test with message about wrong # of args;
}